collections/group_by.ts

```diff
@@ -22,16 +21,16 @@ import { Grouping, Selector } from "./types.ts";
  * ];
  * const peopleByFirstLetter = groupBy(people, it => it.name.charAt(0))
  *
- * console.assert(peopleByFirstLetter === {
+ * assertEquals(peopleByFirstLetter, {
  *     'A': [ { name: 'Anna' }, { name: 'Arnold' } ],
  *     'K': [ { name: 'Kim' } ],
  * })
  * ```
  */
 export function groupBy<T>(
   array: Array<T>,
-  selector: Selector<T, string>,
-): Grouping<T> {
+  selector: (el: T) => string,
+): { [key: string]: Array<T> } {
```

Comment:
Minor nitpick, we might want to use `Record` here since that is what's used throughout `collections`.

Reply:
You are right

Reply:
Done.